crates/node/src/main.rs

```diff
@@ -253,7 +256,10 @@ async fn run(config: Arc<Config>) -> Result<()> {
     let workflow_engine = Arc::new(WorkflowEngine::new(database.clone()));
     let download_url_prefix = format!(
         "http://{}:{}",
-        config.p2p_listen_addr.ip(),
+        config
+            .p2p_advertised_listen_addr
+            .unwrap_or(p2p_listen_addr)
+            .ip(),
```

Comment:
This looks weird. How come it doesn't conflict with `main`: https://github.com/gevulotnetwork/gevulot/blob/main/crates/node/src/main.rs#L257-L264 :thinking:

Reply:
I've already rebase this commit. This modification came from it. Strange, it's still a diff.